Release 18.8.0 (#66534)
gravitational/teleport: CHANGELOG.md

The improvement in memory usage trades off an additional 7MiB of baseline memory usage for a significant reduction in the per-session memory usage of about 23 MiB for each shell or command execution, with another 20 MiB of memory savings for each SSH connection using port forwarding, and about 45 MiB for SFTP sessions.

### VNet for Linux

Teleport VNet support extends to Linux workstations.

### Improvements to access list creation UX

Teleport provides guided in-product UX for creating common types of access lists centered around granting users permissions to resources and permissions to request access to resources.

### tsh MFA via browser

tsh delegates MFA checks (both on login and for per-session MFA) to the browser, enabling the use of browser based passkeys or password managers with tsh.

### Multi-domain support for Windows desktop access

Teleport supports RDP connections to Windows hosts where the Windows users belong to different Active Directory domains than the target hosts.

### Bound keypair joining for agents

Teleport's bound keypair join method extends to support arbitrary Teleport agents in addition to bots.

### Session summaries search

> Identity Security provides users with CLI tooling for searching session summaries allowing users to find sessions based on natural language queries.

Reviewer: Tagging @benarent for review.

If I was an Identity Security user and I read this note I would have questions about what the CLI tooling is and where I can learn more.

Reviewer 2: The docs will live in docs/identity-security/session-summaries/session-search/

https://github.com/gravitational/teleport/pull/66554 is still waiting for +1

Reviewer 3: Let me know if this section is to be updated (i.e link to documentation once https://github.com/gravitational/teleport/pull/66554 is merged).

Reviewer 2: https://goteleport.com/docs/identity-security/session-summaries/session-search/ is live

* Increased verbosity of Teleport Discovery Service logs for VM discovery. [#66553](https://github.com/gravitational/teleport/pull/66553)
* Improved Teleport Connect startup reliability on Windows. [#66509](https://github.com/gravitational/teleport/pull/66509)
* Added Azure Discovery With Terraform integration guided flow in Web UI. [#66493](https://github.com/gravitational/teleport/pull/66493)
* Added a terraform module to support automatic discovery of Azure resources. [#66476](https://github.com/gravitational/teleport/pull/66476)
* Fixed unnecessary double MFA challenge when adding a MFA device using Web UI. [#66474](https://github.com/gravitational/teleport/pull/66474)
* Hardened event handler so it recovers in case of malformed session ID or corrupted data directory. [#66473](https://github.com/gravitational/teleport/pull/66473)

> * Added support for multi-domain sign-on to Windows Desktop Service. [#66471](https://github.com/gravitational/teleport/pull/66471)

Reviewer: This is mentioned above, no need to duplicate it here.

Reviewer 3: updated
removed entire block, just calling out that
`* Hardened event handler so it recovers in case of malformed session ID or corrupted data directory. [#66473](https://github.com/gravitational/teleport/pull/66473)`
was not entirely clear as a duplicate item, just in case we want to add this back.

Reviewer 3: `* Hardened event handler so it recovers in case of malformed session ID or corrupted data directory. [#66473](https://github.com/gravitational/teleport/pull/66473)`
has been added back

Reviewer: Sorry, I didn't mean to select multiple lines here, the only item that needs to be removed is the multi-domain sign-on for Windows desktops.

* Fixed an issue where WebAssembly not being available would crash the web UI. [#66216](https://github.com/gravitational/teleport/pull/66216)
* Added audit events for Azure VM auto-discovery installations, with install script output and exit status. [#66067](https://github.com/gravitational/teleport/pull/66067)
* Fixed an issue where EC2 auto-discovery could install Teleport on an instance but silently drop the failure when the agent could not join the cluster. A new `ec2-join-failure` user task is now raised with the actual join error message surfaced from the agent's readyz socket. [#66023](https://github.com/gravitational/teleport/pull/66023)
* Adds EKS support when using AWS discovery with Terraform. [#65675](https://github.com/gravitational/teleport/pull/65675)
* Added support for `WorkloadIdentity` when using the `--apply-on-startup` and `--bootstrap` flags. [#65581](https://github.com/gravitational/teleport/pull/65581)
* Fixed a bug where tbot's `/readyz` endpoint would report "unhealthy" even after identity renewal succeeds on-retry. [#65258](https://github.com/gravitational/teleport/pull/65258)

> * Added browser-based MFA option to tsh, enabling passkey/biometric authentication via the browser. [#65239](https://github.com/gravitational/teleport/pull/65239)

Reviewer: This can be removed, it is mentioned above.

Reviewer 3: updated
removed entire block, just calling out that
```
* Added support for `WorkloadIdentity` when using the `--apply-on-startup` and `--bootstrap` flags. [#65581](https://github.com/gravitational/teleport/pull/65581)
* Fixed a bug where tbot's `/readyz` endpoint would report "unhealthy" even after identity renewal succeeds on-retry. [#65258](https://github.com/gravitational/teleport/pull/65258)
```
was not entirely clear as a duplicate item, just in case we want to add this back.

Reviewer 3: ```
* Added support for `WorkloadIdentity` when using the `--apply-on-startup` and `--bootstrap` flags. [#65581](https://github.com/gravitational/teleport/pull/65581)
* Fixed a bug where tbot's `/readyz` endpoint would report "unhealthy" even after identity renewal succeeds on-retry. [#65258](https://github.com/gravitational/teleport/pull/65258)
```
have been added back

* Enabled silent automatic updates for Teleport Connect per-machine installations on Windows; elevated privileges are now only required during the initial setup. [#65173](https://github.com/gravitational/teleport/pull/65173)
* Deprecated the `TELEPORT_CDN_BASE_URL` and `TELEPORT_TOOLS_VERSION` environment variables for configuring Teleport Connect Windows updates. These must now be managed via system policy registry keys under `HKEY_LOCAL_MACHINE` or `HKEY_CURRENT_USER\SOFTWARE\Policies\Teleport\TeleportConnect`. The environment variables are still read for compatibility, but per-machine updates may require UAC prompts until configuration is migrated to registry policy keys. [#65173](https://github.com/gravitational/teleport/pull/65173)
* Automatic updates in Teleport Connect no longer allow app version downgrades (applies to all platforms). [#65173](https://github.com/gravitational/teleport/pull/65173)
* Add VNet support for Teleport Connect on Linux. [#64737](https://github.com/gravitational/teleport/pull/64737)

> * Added Linux support for VNet. [#64736](https://github.com/gravitational/teleport/pull/64736)

Reviewer: These two lines can be removed, they're mentioned above.

Reviewer 3: updated

* Added support for reverse tunnel agent stale connection timeout detection and recovery. [#62531](https://github.com/gravitational/teleport/pull/62531)

Enterprise:

> * Reject AWSIC System Credentials on Teleport Cloud.

Reviewer: ```suggestion
* Reject AWS Identity Center System Credentials on Teleport Cloud.
```

Reviewer 3: updated

* Validate AWS Identity Center install credentials with AWS API calls.
* Added support for Terraform configuration generation in the Access List creation wizard in the web UI, allowing users to deploy their Access List via Terraform.

> * Fix a potential deadlock caused by incorrectly reacquiring the pool in the CRDB flow.

Reviewer: ```suggestion
* Fix a potential deadlock in the CockroachDB backend.
```

Reviewer 3: updated

# Changelog

> ## 18.8.0 (05/07/26)

Reviewer 4: For the changelog, we should call out all major deliverables from https://goteleport.com/docs/upcoming-releases/#1880. Take a look at past minor release changelog PRs to see how we usually do it.

The improvement in memory usage trades off an additional 7MiB of baseline memory usage for a significant reduction in the per-session memory usage of about 23 MiB for each shell or command execution, with another 20 MiB of memory savings for each SSH connection using port forwarding, and about 45 MiB for SFTP sessions.

### VNet for Linux

Teleport VNet support extends to Linux workstations.

### Improvements to access list creation UX

Teleport provides guided in-product UX for creating common types of access lists centered around granting users permissions to resources and permissions to request access to resources.

### tsh MFA via browser

tsh delegates MFA checks (both on login and for per-session MFA) to the browser, enabling the use of browser based passkeys or password managers with tsh.

### Multi-domain support for Windows desktop access

Teleport supports RDP connections to Windows hosts where the Windows users belong to different Active Directory domains than the target hosts.

### Bound keypair joining for agents

> Teleport's bound keypair join method extends to support arbitrary Teleport agents in addition to bots.

Reviewer 5: This needs a link to the bound keypair join method, or a guide on how to do it. This should also answer the following user questions:
- Should I use the feature?
- How is it different from the secret token?

Reviewer 5: [docs PR for context](https://github.com/gravitational/teleport/pull/65768), we might want to merge it prior to releasing

Reviewer 6: Merge and backport is in progress, it will be live at https://goteleport.com/docs/installation/agents/bound-keypair/ later today

Preview site for reference: https://timothyb89-bound-keypair-for-agents-docs.d3b94eevwi10ji.amplifyapp.com/docs/installation/agents/bound-keypair/

### tsh MFA via browser

tsh delegates MFA checks (both on login and for per-session MFA) to the browser, enabling the use of browser based passkeys or password managers with tsh.

### Multi-domain support for Windows desktop access

Teleport supports RDP connections to Windows hosts where the Windows users belong to different Active Directory domains than the target hosts.

### Bound keypair joining for agents

Teleport's bound keypair join method extends to support arbitrary Teleport agents in addition to bots.

### Session summaries search

Identity Security provides users with CLI tooling for searching session summaries allowing users to find sessions based on natural language queries.

### Terraform support for AWS EKS discovery

Users are now able to set up AWS EKS discovery on an AWS account level using Terraform module.

### Terraform support for access list workflows

Short and long term access list creation flows in the web UI now include terraform support allowing users to define access with infrastructure-as-code.

> 

Reviewer 7: Could we also add a summary for the Teleport Connect managed updates changes?

```suggestion
### Teleport Connect installation and updates

Teleport Connect for Windows now supports both per-machine and per-user installations. (Note: VNet is not available in per-user mode.)
Per-machine installations can now receive automatic updates without prompting for administrator privileges. Those privileges are only required during the initial installation.

Starting with this release, Teleport Connect only supports automatic upgrades. Downgrades must now be performed manually. This change applies to all platforms.

```

Reviewer 8: Just add that as a commit to the release branch, that's what I was told to do for the SSH internal improvements.

* Enabled silent automatic updates for Teleport Connect per-machine installations on Windows; elevated privileges are now only required during the initial setup. [#65173](https://github.com/gravitational/teleport/pull/65173)
* Deprecated the `TELEPORT_CDN_BASE_URL` and `TELEPORT_TOOLS_VERSION` environment variables for configuring Teleport Connect Windows updates. These must now be managed via system policy registry keys under `HKEY_LOCAL_MACHINE` or `HKEY_CURRENT_USER\SOFTWARE\Policies\Teleport\TeleportConnect`. The environment variables are still read for compatibility, but per-machine updates may require UAC prompts until configuration is migrated to registry policy keys. [#65173](https://github.com/gravitational/teleport/pull/65173)
* Automatic updates in Teleport Connect no longer allow app version downgrades (applies to all platforms). [#65173](https://github.com/gravitational/teleport/pull/65173)
* Add VNet support for Teleport Connect on Linux. [#64737](https://github.com/gravitational/teleport/pull/64737)
* Added Linux support for VNet. [#64736](https://github.com/gravitational/teleport/pull/64736)

> * Adds support for reverse tunnel agent stale connection timeout detection and recovery. [#62531](https://github.com/gravitational/teleport/pull/62531)

Reviewer 9: Can this be made consistent? One "Add", one "Added" and one "Adds". Our changelog requirements state that they should be written in the past tense. Sometimes these slip through on the PRs but then we typically fix them up at this point. There's a couple of others further up too, but these three stand out as a bit messy being one after the other.

Reviewer 5: > Sometimes these slip through on the PRs but then we typically fix them up at this point.

Yeah I find those all the time but I don't think most reviewers enforce the rule. This might be worth another reminder, especially asking G1 reviewers to enforce during review.

Also, could we lint for those? e.g. banning `Add`, `Adds`, and `Fix`, `Fixes` at the beginning of the sentence, or sending into an LLM.

### VNet for Linux

Teleport VNet support extends to Linux workstations.

### Improvements to access list creation UX

Teleport provides guided in-product UX for creating common types of access lists centered around granting users permissions to resources and permissions to request access to resources.

### tsh MFA via browser

tsh delegates MFA checks (both on login and for per-session MFA) to the browser, enabling the use of browser based passkeys or password managers with tsh.

### Multi-domain support for Windows desktop access

Teleport supports RDP connections to Windows hosts where the Windows users belong to different Active Directory domains than the target hosts.

### Bound keypair joining for agents

Teleport's bound keypair join method extends to support arbitrary Teleport agents in addition to bots.

### Session summaries search

Identity Security provides users with CLI tooling for searching session summaries allowing users to find sessions based on natural language queries.

### Reduce resource consumption of SSH sessions

Teleport SSH session latency is reduced and each session consumes less memory.

> 

Reviewer 5: This is a duplicate of the first item

```suggestion
```
